pandas/core/style.py

```diff
@@ -0,0 +1,777 @@
+            result = func(self.data.loc[subset], **kwargs)
+        self._update_ctx(result)
+        return self
+
+    def apply(self, func, axis=0, subset=None, **kwargs):
+        """
+        Apply a function, updating the HTML representation with the result.
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        func: function
+        axis: int, str or None
+            apply to each column (``axis=0`` or ``'index'``)
+            or to each row (``axis=1`` or ``'columns'``) or
+            to the entire DataFrame at once with ``axis=None``.
+        subset: IndexSlice
+            a valid indexer to limit ``data`` to *before* applying the
+            function. Consider using a pandas.IndexSlice
+        kwargs: dict
+            pass along to ``func``
+
+        Returns
+        -------
+        self
+
+        Notes
+        -----
+        This is similar to DataFrame.apply, except that axis=None applies
+        the function to the entire DataFrame at once, rather tha column
+        or rowwise.
+        """
+        self._todo.append((lambda instance: getattr(instance, '_apply'),
+                          (func, axis, subset),
+                           kwargs))
+        return self
+
+    def _applymap(self, func, subset=None, **kwargs):
+        func = partial(func, **kwargs)  # applymap doesn't take kwargs?
+        if subset is None:
+            subset = pd.IndexSlice[:]
+        subset = _non_reducing_slice(subset)
+        result = self.data.loc[subset].applymap(func)
+        self._update_ctx(result)
+        return self
+
+    def applymap(self, func, subset=None, **kwargs):
+        """
+        Apply a function elementwise, updating the HTML
+        representation with the result.
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        func : function
+        subset : IndexSlice
+            a valid indexer to limit ``data`` to *before* applying the
+            function. Consider using a pandas.IndexSlice
+        kwargs : dict
+            pass along to ``func``
+
+        Returns
+        -------
+        self
+
+        """
+        self._todo.append((lambda instance: getattr(instance, '_applymap'),
+                          (func, subset),
+                          kwargs))
+        return self
+
+    def set_precision(self, precision):
+        """
+        Set the precision used to render.
+
+        .. versionadded:: 0.17.1
+
+        Parameters
+        ----------
+        precision: int
+
+        Returns
+        -------
+        self
+        """
+        self.precision = precision
+        return self
+
+    def export(self):
+        """
+        Export the styles to applied to the current Styler.
+        Can be applied to a second style with `.set`.
+
+        .. versionadded:: 0.17.1
+
+        Returns
+        -------
+        styles: list
+        """
+        return self._todo
+
+    def set(self, styles, table_styles=None):
```

Comment:
And the desire to support something like `pd.options.display.style` is why I accept the `table_styles` kwarg in `.set` / `.use`. That way you can bundle things like `:hover` etc. Otherwise you have to `.use('cool').set_table_styles([hover, ...])`. We could package the applied styles + table_styles into some data structure with a name and stuff; just kept it simple for now..